packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback, useContext } from 'react';
```

Comment:
Not a real action point, but would love to hear what @capJavert and @sshanzel think.
To me this file is very hard to follow because of the complexity of hijacking.
I'm not even sure what the alternative is (expect our discussion of uniforming the click, but later on)

Just wondering if it's clear enough to proceed or we can do some simple renaming/refactoring for clarity sake.

Reply:
Understandable though I agree we can probably do some refactoring, such as putting together the `toggleModal` and `activateCompanion` functions in a hook, probably called `useCompanionModal` or something similar. That alone would probably reduce the lines from this file and isolate the actions.

Reply:
I find it ok, more worried over a lot of `eslint-disable` for hook rules, all of those should be resolved before merging.